test/gtest/uct/ib/test_ib.cc

```diff
@@ -440,11 +440,46 @@ class test_uct_ib_sl : public test_uct_ib_with_specific_port {
         }
     }
 
+    bool transport_has_ddp()
+    {
+        ucs::handle<uct_md_h> uct_md;
+
+        if (!has_transport("rc_mlx5") && !has_transport("dc_mlx5")) {
+            return false;
+        }
+
+        UCS_TEST_CREATE_HANDLE(uct_md_h, uct_md, uct_md_close, uct_md_open,
+                               &uct_ib_component,
+                               ibv_get_device_name(m_ibctx->device),
+                               m_md_config);
+
+        uct_ib_mlx5_md_t *ib_md = ucs_derived_of(uct_md, uct_ib_mlx5_md_t);
+        int rc_has_ddp          = has_transport("rc_mlx5") &&
+                                  (ib_md->dp_ordering_cap.rc ==
+                                   UCT_IB_MLX5_DP_ORDERING_OOO_ALL);
+        int dc_has_ddp          = has_transport("dc_mlx5") &&
+                                  (ib_md->dp_ordering_cap.dc ==
+                                   UCT_IB_MLX5_DP_ORDERING_OOO_ALL);
+        return rc_has_ddp || dc_has_ddp;
+    }
+
+    void set_ddp_enable(bool enable)
+    {
+        if (has_transport("rc_mlx5")) {
+            modify_config("RC_MLX5_DDP_ENABLE", enable ? "y" : "n");
```

Comment:
1. if we set just RC_MLX5_DDP_ENABLE, IMO it should enable for both RC and DC, because of inheritance.
2. when "enable=true", don't we want to set to "try"?

Reply:
```
[ RUN      ] dc_mlx5/test_uct_ib_sl.check_ib_sl_config_with_ddp/0 <dc_mlx5/mlx5_0:1>
[     SKIP ] (RC_MLX5_DDP_ENABLE is not a valid configuration)
```

It looks like DC is skipping, meaning we do need to use both variables